js/views/navigationView.js

```diff
@@ -24,17 +26,54 @@ class NavigationView extends Backbone.View {
     };
   }
 
+  get buttons() {
+    return (this._buttons = this._buttons || []);
+  }
+
+  set buttons(value) {
+    this._buttons = value;
+  }
+
   initialize() {
     tooltips.register({
       _id: 'back',
       ...Adapt.course.get('_globals')?._extensions?._navigation?._backNavTooltip || {}
     });
     _.bindAll(this, 'sortNavigationButtons');
+    this._classSet = new Set(_.result(this, 'className').trim().split(/\s+/));
+  }
+
+  start(model) {
+    this.model = model;
+    this.listenTo(model, 'change', this.update);
     this.listenToOnce(Adapt, 'courseModel:dataLoading', this.remove);
-    this.listenTo(Adapt, 'router:menu router:page', this.hideNavigationButton);
+    this.listenTo(Adapt, {
+      'router:contentObject': this.hideNavigationButton,
+      'adapt:preInitialize device:resize': this.onDeviceResize
+    });
+    this.update();
     this.preRender();
   }
 
+  update() {
+    this.updateViewProperties();
+    this.onDeviceResize();
+  }
+
+  onDeviceResize() {
+    let {
+      _navigationAlignment,
+      _isBottomOnTouchDevices
+    } = this.model.toJSON();
+    const $html = $('html');
+    $html.addClass('is-nav-top');
```

Comment:
why is the `is-nav-top` class added here if it is removed immediately below in line 73?

Reply:
Corrected